### PR TITLE
feat(nx-plugin): slim down default generated nx-plugin

### PR DESCRIPTION
**Pull request by AgentEnder**
## Current Behavior
nx g plugin creates:
- e2e tests by default
- Boilerplate executor + generator

## Expected Behavior
- To add e2e you can specify `--e2eTestRunner jest`, or generate it with `nx g @nrwl/nx-plugin:e2e-project`
- You can add a generator with `nx g generator`

## Related Issue(s)

Fixes #

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204424637634825

### CODE REVIEW

**Review by nartc — APPROVED** [review]

**Review by mandarini — APPROVED** [review]

File: packages/nx-plugin/src/generators/executor/schema.json [hunk 1]
```diff
@@ -46,6 +46,11 @@
       "type": "boolean",
       "default": false,
       "description": "Should the boilerplate for a custom hasher be generated?"
+    },
+    "skipLintChecks": {
+      "type": "boolean",
+      "default": false,
+      "description": "Do not eslint configuration for plugin json files."
```
Comment (barbados-clemens) [inline packages/nx-plugin/src/generators/executor/schema.json:53]: this reads kind of funky I think you meant something alone these lines?
```suggestion
      "description": "Do not generate an eslint configuration for plugin json files."
```

**Review by barbados-clemens — APPROVED** [review]

File: packages/nx-plugin/src/generators/e2e-project/files/tests/__pluginName__.spec.ts__tmpl__ [hunk 2]
```diff
@@ -23,39 +24,14 @@ describe('<%= pluginName %> e2e', () => {
...
-  <% } %>
+  // Add some tests here to check that your plugin functionality works as expected.
+  // A sample test is included below to give you some ideas.
+  //
+  // it('should be able to build generated projects', async () => {
+  //   const name = uniq('proj')
+  //   await runNxCommandAsync(`generate <%= npmPackageName %>:{my-generator} --name ${name}`);
+  //   expect(() => runNxCommand('build ${proj}')).not.toThrow();
+  //   expect(() => checkFilesExist([`dist/${name}/index.js`])).not.toThrow();
+  // });
```
Comment (FrozenPandaz) [inline packages/nx-plugin/src/generators/e2e-project/files/tests/__pluginName__.spec.ts__tmpl__:36]: Uncomment and use `xit`

File: packages/nx-plugin/src/generators/e2e-project/files/tests/__pluginName__.spec.ts__tmpl__ [hunk 3]
```diff
@@ -23,39 +24,14 @@ describe('<%= pluginName %> e2e', () => {
...
-      expect(project.tags).toEqual(['e2etag', 'e2ePackage']);
-    }, 120000);
-  });
-  <% } %>
+  // Add some tests here to check that your plugin functionality works as expected.
+  // A sample test is included below to give you some ideas.
+  //
+  // it('should be able to build generated projects', async () => {
+  //   const name = uniq('proj')
+  //   await runNxCommandAsync(`generate <%= npmPackageName %>:{my-generator} --name ${name}`);
```
Comment (FrozenPandaz) [inline packages/nx-plugin/src/generators/e2e-project/files/tests/__pluginName__.spec.ts__tmpl__:33]: `const generator = 'PLACEHOLDER'`

File: packages/nx-plugin/src/generators/generator/generator.ts [hunk 4]
```diff
@@ -81,25 +82,35 @@ function addFiles(host: Tree, options: NormalizedSchema) {
...
     host,
-    joinPathFragments(options.projectRoot, 'generators.json'),
+    joinPathFragments(projectRoot, 'generators.json'),
     {
       generators: {},
     }
   );
+  if (!skipLintChecks) {
+    await pluginLintCheckGenerator(host, {
+      projectName,
```
Comment (FrozenPandaz) [inline packages/nx-plugin/src/generators/generator/generator.ts:108]:
```suggestion
      projectName,
      skipFormat: true
```

File: packages/nx-plugin/src/generators/preset/generator.ts [hunk 5]
```diff
@@ -24,8 +25,13 @@ export default async function (tree: Tree, options: PresetGeneratorSchema) {
     unitTestRunner: 'jest',
     importPath: options.pluginName,
     rootProject: true,
+    e2eTestRunner: 'jest',
   });
 
+  const { root } = readProjectConfiguration(tree, options.pluginName);
+  await createExecutorsJson(tree, root, options.pluginName);
+  await createGeneratorsJson(tree, root, options.pluginName);
```
Comment (FrozenPandaz) [inline packages/nx-plugin/src/generators/preset/generator.ts:33]: Remove this.

**Review by FrozenPandaz — APPROVED** [review]

File: packages/nx-plugin/src/utils/has-generator.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  );
+  if (!packageJson.generators && !packageJson.schematics) {
+    return false;
+  }
+  const generatorsPath = joinPathFragments(
+    project.root,
+    packageJson.generators ?? packageJson.schematics
+  );
+  const generatorsJson = readJson<GeneratorsJson>(tree, generatorsPath);
+  return generatorsJson.generators[generatorName] !== undefined;
```
Comment (FrozenPandaz) [inline packages/nx-plugin/src/utils/has-generator.ts:28]: Check schematics too

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**